src/main/java/de/zalando/aruha/nakadi/exceptions/NakadiException.java

```diff
@@ -26,4 +31,11 @@ public void setProblemMessage(final String problemMessage) {
         this.problemMessage = problemMessage;
     }
 
+    public Problem asProblem() {
+        final String detail = StringUtils.isNotBlank(problemMessage) ? problemMessage : getMessage();
+        return Problem.valueOf(getStatus(), detail);
+    }
+
+    protected abstract Response.StatusType getStatus();
```

Comment:
I think from a client code perspective this approach worked just perfectly. But I was just wondering if isn't it conceptually broken. The exceptions are "domain specific", e.g, they are not aware that this code is running in the context of an http request.

I don't think it's going to harm us in the future, but I just found it strange that a Domain exception knows about the Http stack.